Ensembl/ensembl-web-track-api
# Optimise Track API datamodels, add track submit endpoint

- Optimise Django datamodels (see [database diagram](https://lucid.app/lucidchart/5cee5c19-5c63-4ba5-b774-7c75e8af4319/edit?viewport_loc=129%2C209%2C1928%2C948%2C0_0&invitationId=inv_56eaae93-6782-4471-a8a8-7a6a661fadee)).
- Implement POST/DELETE in `/track` endpoint to add or remove tracks

Review app: http://submit-track.review.ensembl.org/api/tracks/
To test: 
- load the api spec in [swagger](https://editor.swagger.io)
- submit a track: `curl http://submit-track.review.ensembl.org/api/tracks/track -H 'Content-Type: application/json' -X POST -d '{...example POST JSON from spec, change e.g. label field...}`
- check the new track (with browser or curl): `http://submit-track.review.ensembl.org/api/tracks/track/:track_id` (id from POST response) or `track_categories/:genome_id` (id from POST track JSON)
- optional: delete the track with `curl -X DELETE http://submit-track.review.ensembl.org/api/tracks/track_categories/:genome_id`

The review app is wired to test postgres database. You can check its content in `kxd` (`postgres-trackdb...` pod)

JIRA:[ ENSWBSITES-2207](https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-2207)
